manifests/validate_db_connection.pp

```diff
@@ -86,9 +86,10 @@
   }
 
   $exec_name = "validate postgres connection for ${database_username}@${database_host}:${database_port}/${database_name}"
+  $exec_command = ['echo', "'Unable to connect to defined database using:", join([$cmd, "' && false"], '')]
```

Comment:
The `&& false` means a shell is used. I'm not sure how it would work, but I think that's just going to fail if you pass it as an array.

Instead, I think the `validate_cmd` should fail in a way that is clear. We control the script so why can't that have a proper echo and exit code?

Reply:
Have put it back to a single string.
As to the second part, that may be outside the scope for this.

Reply:
I wonder if these days you may even be able to implement it as a deferred function.

Reply:
Maybe, there has been some talk of deferred functions lately.